packages/keycloak-user-management/src/components/CreateEditUser/index.tsx

```diff
@@ -36,112 +38,112 @@ export interface RouteParams {
+      serve
+        .list()
+        .then((response: UserGroup[]) => setUserGroups(response))
+        .catch((_: Error) => sendErrorNotification(ERROR_OCCURED));
+    }
+  }, [keycloakBaseURL, opensrpBaseURL, userGroups.length]);
 
   /**
    * Fetch user incase the user is not available e.g when page is refreshed
    */
-  React.useEffect(() => {
+  useEffect(() => {
     if (userId && !keycloakUser) {
-      const serve = new serviceClass(KEYCLOAK_URL_USERS, keycloakBaseURL);
-      setIsLoading(true);
+      const serve = new KeycloakService(KEYCLOAK_URL_USERS, keycloakBaseURL);
       serve
         .read(userId)
         .then((response: KeycloakUser | null | undefined) => {
-          if (response) {
-            setIsLoading(false);
-            fetchKeycloakUsersCreator([response]);
-          }
-        })
-        .catch((_: Error) => {
-          sendErrorNotification(ERROR_OCCURED);
+          if (response) fetchKeycloakUsers([response]);
         })
-        .finally(() => setIsLoading(false));
+        .catch((_: Error) => sendErrorNotification(ERROR_OCCURED));
+    }
+  }, [userId, keycloakBaseURL, keycloakUser]);
+
+  /**
+   * Fetch User group of the user
+   */
+  useEffect(() => {
+    if (userId && initialValues.userGroup === undefined) {
+      const serve = new KeycloakService(
+        KEYCLOAK_URL_USERS + '/' + userId + KEYCLOAK_URL_USER_GROUPS,
+        keycloakBaseURL
+      );
+      serve
+        .list()
+        .then((response: UserGroup[]) =>
+          setInitialValues({ ...initialValues, userGroup: response.map((tag) => tag.id) })
+        )
+        .catch((_: Error) => sendErrorNotification(ERROR_OCCURED));
     }
-  }, [fetchKeycloakUsersCreator, serviceClass, userId, keycloakBaseURL, keycloakUser]);
+  }, [userId, keycloakBaseURL, initialValues]);
 
-  React.useEffect(() => {
-    if (userId && practitioner === undefined) {
-      setIsLoading(true);
-      const serve = new opensrpServiceClass(OPENSRP_CREATE_PRACTITIONER_ENDPOINT, opensrpBaseURL);
+  useEffect(() => {
+    if (userId && initialValues.practitioner === undefined) {
+      const serve = new OpenSRPService(OPENSRP_CREATE_PRACTITIONER_ENDPOINT, opensrpBaseURL);
       serve
         .read(userId)
-        .then((response: Practitioner) => {
-          setIsLoading(false);
-          setPractitioner(response);
-        })
-        .catch((_: Error) => {
-          sendErrorNotification(ERROR_OCCURED);
-        })
-        .finally(() => setIsLoading(false));
+        .then((response: Practitioner) =>
+          setInitialValues({ ...initialValues, active: response.active, practitioner: response })
+        )
+        .catch((_: Error) => sendErrorNotification(ERROR_OCCURED));
     }
-  }, [userId, practitioner, opensrpServiceClass, opensrpBaseURL]);
+  }, [userId, opensrpBaseURL, initialValues]);
 
-  if (isLoading) {
+  if (!userGroups.length || (keycloakUser && !initialValues.username) || (userId && !keycloakUser))
```

Comment:
does this mean, we get to show an eternal spinner if one of the above is truthy, Ideally, we should show the loader, only when there is some loading happening.

Reply:
yes we get to show that spinner when any of that or is happening and that or is happening only when loading that the and these condition are must for the sub component initialization and mount

Reply:
@ashfahan Do you think it would have been better to check `isLoading` rather than checking all these conditions?

Reply:
it was a bit harder to manage isloading state that is why i did this @kelvin-muchiri